### DISCUSSION TITLE
Trouble with Closure

### DISCUSSION THREAD

**Discussion by Alex-128** [saeg/ba-dua#42] [Q&A]
Hi, me again! :)

I've been able to use BA-DUA for data-flow coverage SBFL on the standard Defects4J projects (Chart, Lang, Math, Time), but with Closure it becomes so slow that it's unusable. I see that in the articles on Jaguar, Closure wasn't included as a benchmark project. Did the same sort of problem occur? Is there a way to fix it? I also saw that you used a subset of faulty versions from your benchmarks, specifically the versions with complete data-flow spectra. Do you have the specification of complete faulty versions available somewhere? Thank you for all the help you've already given me!

Kind regards
Alexandra van der Spuy

**Comment by andrioli** [comment]
Hi Alexandra,

Digging on my emails I didn't find a reason why Closure was not include as a benchmark project. I'll ask the other group members to check if anyone recall why and if someone have the specification of complete faulty versions.

But also on my emails I found a discussion about Closure running slow and unstable. The discussion conclusion was high memory consumption and high GC overhead. In the same discussion we suggested to increase the maximum Java heap size (e.g., `-Xmx1536m`) and looks that this fixed the problem. Can you test it ?

**Comment by Alex-128** [comment]
Hi Roberto

Sorry for the long silence, but I managed to figure out this issue with Closure. Unfortunately, I now have a new one! 

When running certain test cases over the instrumented Closure classes, I get the following error:
```
java.lang.ClassFormatError: StackMapTable format error: bad offset for Uninitialized in method com.google.javascript.jscomp.PeepholeFoldConstants.tryFoldComparison(Lcom/google/javascript/rhino/Node;Lcom/google/javascript/rhino/Node;Lcom/google/javascript/rhino/Node;)Lcom/google/javascript/rhino/Node;
```
This only happens with a few classes, and only with Closure. It seems to be an issue on BA-DUA's side, because it doesn't happen when the class (in this case, `com.google.javascript.jscomp.PeepholeFoldConstants`) is uninstrumented while other classes are instrumented. Have you or your fellow developers encountered this issue before? Do you have any advice on what could be causing this?

Kind regards
Alexandra
